app/models/aggregated_cash_income.rb

```diff
@@ -0,0 +1,106 @@
+class AggregatedCashIncome
+  include ActiveModel::Model
+
+  CATEGORIES= %i[
+      benefits
+      friends_or_family
+      maintenance_in
+      property_or_lodger
+      pension
+  ].freeze
+
+  RANGE = (1..3).freeze
+
+  def self.category_attributes(categories)
+    categories.each do |category|
+      attr_accessor "check_box_#{category}".to_sym
+      RANGE.each do |i|
+        attr_accessor "#{category}_month_#{i}".to_sym
+      end
+    end
+  end
+
+  attr_accessor :legal_aid_application_id,
+                :category,
+                :none_selected
+
+  category_attributes CATEGORIES
+
+  validate :check_category
+  validate :anything_selected?
+
+  def checkbox_integrity(params)
+    CATEGORIES.each do |category|
+      checkbox_selected = params["check_box_#{category}"] == 'true'
+
+      if checkbox_selected
+        @category = category
+        check_category
+      else
+        args = search_args(category)
+        CashTransaction.where(**args).destroy_all
+      end
+    end
+
+    anything_selected?
+  end
+
+  def save(params)
+    params.each do |key, value|
+      __send__("#{key}=",value) if value != ''
+    end
+
+    checkbox_integrity(params)
+
```

Comment:
I think rather than call a specific validation method here, we should just call `valid?` which will call ALL the validations specified in the model.  So here we could write 
``` 
return false unless valid? 
```
instead of lines 53 and 55.

Reply:
So interestingly I tried that and it didn't work, for now I suck in errors.empty? but once I've tidied up, if I can't get that working I'll run it by you. I think valid? might reset itself after adding the first failed validation or something as the frontend was only displaying 1 error instead of 3 for each field. Anyway I'll give the above a try. It might be we are missing something.